FilePersistence/src/version_control/Commit.cpp

```diff
@@ -82,8 +82,13 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 	else
 	{
 		iter = files_.constBegin();
-		while (!iter.key().contains(fileName) && iter != files_.constEnd())
+		while (iter != files_.constEnd())
```

Comment:
Hm, I already avoid matching directories by using `QFileInfo::fileName()` which returns just the filename without the path.

The check afterwards is to see if it is an UID which can be matched using contains or something else, where contains would not be good (file "foobar" contains "foo") and thus I use exact matching for just the filename.

But I agree it is brittle.